packages/create-react-app/index.js

```diff
@@ -349,6 +367,9 @@ function moveReactScriptsToDev(packageName) {
   packageJson.devDependencies[packageName] = packageVersion;
   delete packageJson.dependencies[packageName];
 
+  packageJson.dependencies['react'] = patchReactDependencyVersion('react', packageJson.dependencies['react']);
```

Comment:
Maybe we could write a mutating `makeCaretRange(dependencies, name)`?
Seems like that would make the signature and both calls simpler.